lib/pbench/server/api/resources/users_api.py

```diff
@@ -0,0 +1,534 @@
+            abort(500, message="INTERNAL ERROR")
+        if user:
+            self.logger.warning("A user tried to re-register. Email: {}", user.email)
+            abort(403, message="Provided email is already in use")
+
+        first_name = user_data.get("first_name")
+        if not first_name:
+            self.logger.warning("Missing firstName field")
+            abort(
+                400, message="Missing firstName field",
+            )
+
+        last_name = user_data.get("last_name")
+        if not last_name:
+            self.logger.warning("Missing lastName field")
+            abort(
+                400, message="Missing lastName field",
+            )
+
+        try:
+            user = User(
+                username=username,
+                password=password,
+                first_name=first_name,
+                last_name=last_name,
+                email=email_id,
+            )
+
+            # insert the user
+            user.add()
+            self.logger.info(
+                "New user registered, username: {}, email: {}", username, email_id
+            )
+
+            response_object = {
+                "status": "success",
+                "message": "Successfully registered.",
+            }
+            response = jsonify(response_object)
+            response.status_code = 201
+            return make_response(response, 201)
+        except EmailNotValidError:
+            self.logger.warning("Invalid email {}", email_id)
+            abort(400, message=f"Invalid email: {email_id}")
+        except Exception:
+            self.logger.exception("Exception while registering a user")
+            abort(500, message="INTERNAL ERROR")
+
+
+class Login(Resource):
+    """
+    Pbench API for User Login or generating an auth token
+    """
+
+    def __init__(self, config, logger, auth):
+        self.server_config = config
+        self.logger = logger
+        self.auth = auth
+        try:
+            self.token_expire_duration = self.server_config.get(
+                "pbench-server", "token_expiration_duration"
+            )
+        except NoOptionError:
+            self.token_expire_duration = None
+
+    @Auth.token_auth.login_required(optional=True, f=Auth().verify_auth())
+    def post(self):
+        """
+        Post request for logging in user.
+        The user is allowed to re-login multiple times and each time a new valid auth token will be provided
+
+        This requires a JSON data with required user metadata fields
+        {
+            "username": "username",
+            "password": "password",
+        }
+
+        Required headers include
+
+            Content-Type:   application/json
+            Accept:         application/json
+
+        :return: JSON Payload
+        if we succeed to decrypt the password hash, the returned response_object will include the auth_token
+            response_object = {
+                    "status": "success", # will not present if failed
```

Comment:
I guess status message is just not very useful, I should probably just remove it. It can also be fetched using `response.status_code` instead of explicitly reading the status message

Reply:
fixed